pkg/scheduler/apis/config/types_pluginargs.go

```diff
@@ -84,6 +84,9 @@ type NodeResourcesFitArgs struct {
 	// with "example.com", such as "example.com/aaa" and "example.com/bbb".
 	// A resource group name can't contain '/'.
 	IgnoredResourceGroups []string
+
+	// ScoringStrategy selects the node resource scoring strategy.
+	ScoringStrategy *ScoringStrategy
```

Comment:
ok, mark this as todo.

Reply:
> When not set, this should default to "LeastAllocated".

-1 on defaulting. Otherwise, it's impossible to achieve "the user doesn't want any resource scoring plugin".

Reply:
We need to maintain the current default scheduler behavior for backward compatibility. If the user doesn't want any scoring, then they can simply disable the scoring plugin.

Reply:
> then they can simply disable the scoring plugin.

If the user disables all resource scoring plugins (explicitly) and we do the defaulting, isn't it interpreted to "LeastAllocated" plugin gets enabled?

Reply:
The defaulting we are referring to here is for the PluginConfig. This defaulting is not going to take an effect if the plugin itself is disabled (listed in the [Disabled slice](https://github.com/kubernetes/kubernetes/blob/4d4b530114a6a160c0366673c73de607eeb54453/pkg/scheduler/apis/config/types.go#L215))

Reply:
ok, the disabled semantic will serve as the final gate.

If we do the defaulting here, and user specifies a (legacy) LeastAllocated plugin, what will happen? I guess this plugin will be semantically called twice?

Reply:
This new plugin and LeastAllocated plugin shouldn't exist at the same time. We will not have this plugin in v1beta1, and we will not have LeastAllocated in v1beta2.

Reply:
That will be the most ideal way. Let's review the conversion logic in the follow-up PR.

Reply:
just to close the loop on this one, we will have this plugin in both v1beta1 and v1beta2.